Runtime/Model/BacktraceHttpClient.cs

```diff
@@ -124,17 +125,39 @@ private List<IMultipartFormSection> CreateMinidumpFormData(byte[] minidump, IEnu
 
         private void AddAttachmentToFormData(List<IMultipartFormSection> formData, IEnumerable<string> attachments)
         {
+            if (attachments == null)
+            {
+                return;
+            }
             // make sure attachments are not bigger than 10 Mb.
             const int maximumAttachmentSize = 10000000;
             const string attachmentPrefix = "attachment_";
-            foreach (var file in attachments)
+
+            var uniqueAttachments = new HashSet<string>(attachments.Reverse());

```

Comment:
Unfortunatelly we still need to make a copy of them because we need to make sure we're using hashset - since someone can add attachment via beforeSend event or we read attachments via database.